packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1366,6 +1372,9 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
     _updateRouting();
     _locale = _resolveLocales(WidgetsBinding.instance.platformDispatcher.locales, widget.supportedLocales);
     WidgetsBinding.instance.addObserver(this);
+    if (WidgetsBinding.instance.lifecycleState != null) {
```

Comment:
Is this check needed?

Reply:
Good call, I'll change the variable to non-nullable and keep this check.

Reply:
Actually, I removed the check and kept it non-nullable, as you suggested.  Otherwise there's a chance it could still be uninitialized.